src/x/query.rs

```diff
@@ -10,6 +10,49 @@ use std::fmt;
 pub trait Query<X: XConn> {
     /// Run this query for a given window ID.
     fn run(&self, id: Xid, x: &X) -> Result<bool>;
+
+    /// Combine this query with another query using a logical AND.
+    ///
+    /// NOTE: This follows typical short-circuiting behavior, i.e. if the first query
+    /// returns false, the second query will not be run.
+    fn and<Other>(self, other: Other) -> AndQuery<X>
+    where
+        Self: Sized + 'static,
+        Other: Query<X> + 'static,
```

Comment:
The `'static` bounds here don't matter, since we are intending for this to take ownership of `self`. Since owned objects are `'static`, this shouldn't be a problem. 

But let me know if you can see an issue with this.

Reply:
No this all looks fine :+1: